akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```diff
@@ -775,6 +802,102 @@ public Integer apply(Integer elem) {
     future.toCompletableFuture().get(200, TimeUnit.MILLISECONDS);
   }
 
+  @Test
+  public void mustBeAbleToRecoverWithClass() throws Exception {
+    final TestPublisher.ManualProbe<Integer> publisherProbe = TestPublisher.manualProbe(true,system);
+    final TestKit probe = new TestKit(system);
+    final Iterable<Integer> recover = Arrays.asList(55, 0);
+
+    final Source<Integer, NotUsed> source = Source.fromPublisher(publisherProbe);
+    final Flow<Integer, Integer, NotUsed> flow = Flow.of(Integer.class)
+        .map(elem -> {
+          if (elem == 2) throw new RuntimeException("ex");
+          else return elem;
+        })
+        .recoverWith(
+          RuntimeException.class,
+          Source.from(recover));
+
+    final CompletionStage<Done> future =
+        source.via(flow).runWith(Sink.foreach(elem -> probe.getRef().tell(elem, ActorRef.noSender())), materializer);
+
+    final PublisherProbeSubscription<Integer> s = publisherProbe.expectSubscription();
+
+    s.sendNext(0);
+    probe.expectMsgEquals(0);
+    s.sendNext(1);
+    probe.expectMsgEquals(1);
+    s.sendNext(2);
+    probe.expectMsgEquals(55);
+    probe.expectMsgEquals(0);
+    future.toCompletableFuture().get(200, TimeUnit.MILLISECONDS);
```

Comment:
increase those to 3000, no reason to have it tight
see https://github.com/akka/akka/blob/master/CONTRIBUTING.md#preferred-ways-to-use-timeouts-in-tests

Reply:
Yes, get it, I will fix it later and thanks.